### DISCUSSION TITLE
Add vcpkg into the 'PATH' variable.

### DISCUSSION THREAD

**Discussion by Mr-Leshiy** [microsoft/vcpkg#19071] [Ideas]
Are you looking for to add vcpkg into 'PATH' env var during the bootstrapping process ?

**Comment by JackBoosY** [comment]
I don't think we need that, @ras0219 @ras0219-msft what do you think about?

**Comment by vicroms** [comment]
Maybe something like `bootstrap-vcpkg.bat -AddToPath` or a post-installation command like `vcpkg integrate PATH` could be implemented.

**Comment by Mr-Leshiy** [comment]
As for me the best solution to do it during the bootstrap process, just to modify windows and linux scripts with the additional argument. The way to update vcpkg directly seems more complicated.

**Comment by BillyONeal** [comment]
Given the dependence on the particular vcpkg tree in use I don't believe this is something we intend to investigate in the medium term; that doesn't mean we're opposed to it in concept. Note that registries will have an effect here.

**Comment by soroshsabz** [comment]
@JackBoosY  I think it is good when we call integrate command, I think in integration command we need to add vcpkg to global system path.

**Comment by Mr-Leshiy** [comment]
@JackBoosY , @BillyONeal  Can I try to suggest an implementation of such feature, if the core team focused on the another things ? I mean will it be good if such functionality will be developed by the community ?

**Comment by BillyONeal** [comment]
> @JackBoosY , @BillyONeal Can I try to suggest an implementation of such feature, if the core team focused on the another things ? I mean will it be good if such functionality will be developed by the community ?

As I mentioned here, the team is indeed not interested in investing in this right now:

> Given the dependence on the particular vcpkg tree in use I don't believe this is something we intend to investigate in the medium term; that doesn't mean we're opposed to it in concept. Note that registries will have an effect here.

If someone submits a PR in the vcpkg-tool repo that implements such changes I'm not saying that we would reject it out of hand. However, I am saying that we believe it is going to be substantially tricky and that there needs to be substantial user experience design here that we don't see most contributors attempt. Without substantial demonstration of consideration of those edge cases it would be a long time before we could consider merging such a PR because that's where the complexity comes from.

You can always manipulate the path in your own shell :)

**Reply by soroshsabz** [reply]
I think setup procedure must be consider it. automation is good. ;)

**Reply by autoantwort** [reply]
Imho it would make sense in the `vcpkg integrate bash|zsh|fish` command because they also set the `VCPKG_ROOT`, so it would make sense to me to also add vcpkg to the path there

**Reply by kotx** [reply]
> Imho it would make sense in the `vcpkg integrate bash|zsh|fish` command because they also set the `VCPKG_ROOT`, so it would make sense to me to also add vcpkg to the path there

All of those shells use PATH, right? So it wouldn't be necessary to specify the shell, TMK.

**Reply by autoantwort** [reply]
But that are existing commands that you can extend

**Reply by alvarogalloc** [reply]
I made a workaround making an alias in my .zshrc as the following:
`alias vcpkg=/Users/macuser/vcpkg/vcpkg`
with that, all features work, including completion

**Reply by soroshsabz** [reply]
I think `vcpkg integrate` must do it
